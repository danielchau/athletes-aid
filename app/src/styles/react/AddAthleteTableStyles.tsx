import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

/**
 * React styles for the Add Athletes Table.
 */
export const addAthleteTableStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: "100%",
            padding: "8px",
            height: "100%",
            display: "flex",
            justifyContent: "center"
        },
        tableContainer: {
            width: "100%",
            height: "100%",
            padding: "8px",
            backgroundColor: "rgba(0, 33, 69, 0.05)",
            overflow: "scroll"
        },
        table: {
            width: "100%"
        },
        tableBodyContainer: {
            width: "2000px",
            overflow: "auto",
            height: "100%"
        },
        tableBody: {
            tableLayout: "fixed"
        },
        tableRow: {
            cursor: "pointer"
        },
        tooltip: {
            backgroundColor: "#fff",
            color: "#4f4f4f",
            boxShadow: theme.shadows[1]
        }
    })
);
