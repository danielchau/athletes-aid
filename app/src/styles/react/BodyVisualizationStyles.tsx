import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

/**
 * React styles for the Body Visualization.
 */
export const bodyVisualizationStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: "100%",
            height: "100%"
        },
        imageContainer: {
            position: "relative",
            alignSelf: "center"
        },
        image: {
            width: "100%"
        },
        dot: {
            height: "15px",
            width: "15px",
            position: "absolute",
            borderRadius: "15px"
        },
        legendContainer: {
            borderLeft: "#dedede solid 1px"
        },
        legendItem: {
            display: "flex",
            color: "rgba(0, 0, 0, 0.54)",
            marginBottom: "2px"
        },
        dotLegend: {
            height: "10px",
            width: "10px",
            borderRadius: "10px",
            marginTop: "6px",
            marginRight: "4px"
        }
    })
);
