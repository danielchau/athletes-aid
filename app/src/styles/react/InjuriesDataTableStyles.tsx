import { makeStyles, Theme, createStyles, lighten } from "@material-ui/core/styles";

/**
 * React styles for the Injury Data Table.
 */
export const injuriesDataTableStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: "100%",
            marginTop: theme.spacing(3)
        },
        paper: {
            width: "100%",
            marginBottom: theme.spacing(2),
            backgroundColor: "rgba(0, 33, 69, 0.05)"
        },
        table: {
            minWidth: 750
        },
        tableWrapper: {
            overflowX: "auto"
        },
        tableRow: {
            cursor: "pointer"
        },
        visuallyHidden: {
            border: 0,
            clip: "rect(0 0 0 0)",
            height: 1,
            margin: -1,
            overflow: "hidden",
            padding: 0,
            position: "absolute",
            top: 20,
            width: 1
        }
    })
);

/**
 * React styles for the Injury Data Table Toolbar Header.
 */
export const injuriesDataTableToolbarStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            paddingLeft: theme.spacing(2),
            paddingRight: theme.spacing(1)
        },
        highlight:
            theme.palette.type === "light"
                ? {
                      color: theme.palette.primary.main,
                      backgroundColor: lighten(theme.palette.primary.light, 0.85)
                  }
                : {
                      color: theme.palette.text.primary,
                      backgroundColor: theme.palette.secondary.dark
                  },
        title: {
            flex: "1 1 100%"
        },
        export: {
            margin: "auto",
            color: theme.palette.secondary.main,
            display: "flex"
        }
    })
);
