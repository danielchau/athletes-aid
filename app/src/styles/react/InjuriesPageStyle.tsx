import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

/**
 * React styles for the Injuries Page.
 */
export const injuriesPageStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            padding: "16px",
            backgroundColor: "#fff"
        },
        drawerOpen: {
            width: "calc(100vw - 240px)",
            transition: theme.transitions.create("width", {
                easing: theme.transitions.easing.sharp,
                duration: theme.transitions.duration.enteringScreen
            }),
            [theme.breakpoints.down("xs")]: {
                width: "100vw"
            }
        },
        drawerClosed: {
            width: "calc(100vw - 72px)",
            transition: theme.transitions.create("width", {
                easing: theme.transitions.easing.sharp,
                duration: theme.transitions.duration.leavingScreen
            }),
            [theme.breakpoints.down("xs")]: {
                width: "100vw"
            }
        },
        grid: {
            backgroundColor: "#fff"
        },
        paper: {
            padding: theme.spacing(2),
            textAlign: "center",
            color: theme.palette.text.secondary,
            backgroundColor: "rgba(0, 33, 69, 0.05)",
            height: "100%"
        },
        vizPaper: {
            padding: theme.spacing(2),
            height: "100%"
        },
        primaryStatisticContainer: {
            display: "flex",
            flexDirection: "column"
        },
        primaryStatisticValue: {
            overflow: "hidden",
            textOverflow: "ellipsis",
            fontSize: "calc((100vw - 240px)/16)"
        },
        primaryStatisticLabel: {
            paddingTop: "16px"
        },
        dateTimeContainer: {},
        textField: {
            marginLeft: theme.spacing(1),
            marginRight: theme.spacing(1),
            width: 200
        },
        dateField: {
            width: 200,
            marginRight: theme.spacing(1),
            marginTop: 0,
            marginBottom: 0
        },
        dateTimeButton: {
            marginLeft: "8px",
            marginTop: "6px"
        }
    })
);
