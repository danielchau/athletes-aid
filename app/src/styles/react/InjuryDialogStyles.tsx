import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

export const injuryDialogStyles = makeStyles((theme: Theme) =>
    createStyles({
        appBar: {
            position: "relative"
        },
        title: {
            marginLeft: theme.spacing(2),
            flex: 1
        },
        dialogTitle: {
            padding: "0"
        },
        dialogPaper: {
            height: "100%",
            maxWidth: "600px",
            width: "50%",
            minWidth: "200px",
            position: "fixed",
            bottom: "0",
            right: "0",
            margin: "0"
        },
        dialogContentContainer: {
            padding: "0"
        },
        dialogContent: {
            padding: "16px"
        },
        dialogContentPaper: {
            backgroundColor: "#fafafa",
            padding: "8px",
            marginBottom: "8px"
        },
        notePaper: {
            backgroundColor: "#ffffff",
            padding: "8px",
            marginBottom: "8px"
        },
        noteContainer: {
            display: "flex",
            flexDirection: "row"
        },
        noteIcon: {
            margin: "auto",
            marginLeft: "8px",
            marginRight: "16px"
        },
        newNoteContainer: {
            display: "flex",
            flexDirection: "row",
            width: "100%",
            justifyContent: "center",
            alignItems: "center",
            alignSelf: "center"
        },
        newNote: {
            flexGrow: 1,
            paddingRight: "8px"
        },
        sendButton: {
            color: "#F2A71E"
        },
        activeSwitchContainer: {
            display: "flex",
            flexDirection: "row",
            justifyContent: "center",
            alignContent: "center",
            alignSelf: "center",
            paddingBottom: "8px"
        }
    })
);
