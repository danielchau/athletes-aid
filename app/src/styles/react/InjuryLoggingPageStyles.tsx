import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

export const injuryLoggingPageStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: "100%",
            padding: "8px"
        },
        backButton: {
            marginRight: theme.spacing(1)
        },
        completedContainer: {
            height: "calc(100% - 116px)",
            textAlign: "center",
            paddingTop: "20%"
        },
        instructions: {
            marginTop: theme.spacing(1),
            marginBottom: theme.spacing(1)
        },
        paper: {
            color: theme.palette.text.secondary,
            backgroundColor: "#fafafa"
        },
        paperContent: {
            height: "100%",
            padding: "16px",
            paddingRight: "8px",
            overflow: "auto"
        },
        stepper: {
            backgroundColor: "#fafafa",
            borderRadius: "4px"
        },
        loggingContent: {
            height: "calc(100% - 168px)",
            marginBottom: "52px",
            marginTop: "8px",
            overflow: "hidden"
        },
        loggingBottomButtons: {
            position: "fixed",
            bottom: 0,
            right: 0,
            height: "52px",
            padding: "8px"
        },
        multilineInput: {
            width: "100%"
        }
    })
);
