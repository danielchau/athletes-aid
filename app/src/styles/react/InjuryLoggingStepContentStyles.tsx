import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

export const injuryLoggingStepContentStyles = makeStyles((theme: Theme) =>
    createStyles({
        formControl: {
            minWidth: 200,
            width: "50%",
            paddingRight: "8px",
            paddingBottom: "16px"
        },
        textInput: {
            minWidth: 200,
            width: "100%",
            paddingRight: "8px",
            paddingBottom: "16px"
        },
        multilineInput: {
            width: "100%",
            paddingBottom: "16px",
            margin: "0px"
        },
        dateInput: {
            minWidth: 200,
            width: "50%",
            margin: "0px",
            paddingRight: "8px",
            paddingBottom: "16px"
        },
        checkboxInput: {
            minWidth: 200,
            width: "50%",
            margin: "0px",
            paddingRight: "8px",
            paddingBottom: "23px",
            paddingTop: "7px"
        },
        instructions: {
            textAlign: "center",
            marginBottom: theme.spacing(3),
            color: theme.palette.text.secondary,
            width: "100%"
        }
    })
);
