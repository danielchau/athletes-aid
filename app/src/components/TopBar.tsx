import * as React from "react";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";
import ChevronLeftIcon from "@material-ui/icons/ChevronLeft";
import SearchIcon from "@material-ui/icons/Search";
import { NavigationPanelStates, Team, Athlete, User } from "../util/types";
import { topBarStyles } from "../styles/react/TopBarStyle";
import { TextField } from "@material-ui/core";
import Autocomplete, { RenderOptionState } from "@material-ui/lab/Autocomplete";
import { Link } from "react-router-dom";
// @ts-ignore
import Logo from "../util/logo.png";
import { profilePath } from "../constants/constants";

interface TopBarProps {
    state: NavigationPanelStates;
    handleDrawerOpen: any;
    handleDrawerClose: any;
    selectedTeam: Team;
    setSelectedAthlete: (id: string) => void;
    currentUser: User;
    currentRoster: Athlete[];
    getCurrentRoster: (athleteIds: string[]) => Promise<Athlete[]>;
}

/**
 * Top Bar is a consistent component of the application that is on the very top.
 * It has the application logo and name and a search bar to search for athletes on the top right.
 * @param props
 */
export default function TopBar(props: TopBarProps) {
    const classes = topBarStyles({});
    const [autocompleteValue, setAutocompleteValue] = React.useState<string>("");
    const [autocompleteOpen, setAutocompleteOpen] = React.useState<boolean>(false);
    const [isFetching, setIsFetching] = React.useState<boolean>(false);
    const [isFirstRender, setIsFirstRender] = React.useState<boolean>(true);

    React.useEffect(() => {
        if (isFirstRender) {
            setIsFirstRender(false);
        } else {
            if (
                !!props.currentRoster &&
                JSON.stringify(props.currentRoster.map(a => a.id).sort()) !=
                    JSON.stringify(props.selectedTeam.athleteIds.sort())
            ) {
                setIsFetching(true);
                props.getCurrentRoster(props.selectedTeam.athleteIds).then(_ => {
                    setIsFetching(false);
                });
            }
        }
    }, [props.selectedTeam]);

    const onAutocompleteInputChange = (event: any, value: string) => {
        if (!!event) {
            if (event.type == "blur") {
                setAutocompleteOpen(false);
            } else if (event.type == "click" || (event.type == "keydown" && event.keyCode == 13)) {
                let selectedAthlete = props.currentRoster.filter(a => value == a.name);
                if (selectedAthlete.length > 0) {
                    props.setSelectedAthlete(selectedAthlete[0].id);
                }
            } else {
                if (!!!props.currentRoster && !!props.selectedTeam) {
                    setIsFetching(true);
                    props.getCurrentRoster(props.selectedTeam.athleteIds).then(_ => {
                        setIsFetching(false);
                    });
                }
                if (autocompleteValue != value) {
                    setAutocompleteValue(value);
                    if (value != "" && !autocompleteOpen) {
                        setAutocompleteOpen(true);
                    } else if (value == "" && autocompleteOpen) {
                        setAutocompleteOpen(false);
                    }
                }
            }
        }
    };

    return (
        <>
            <AppBar position="fixed" className={classes.appBar}>
                <Toolbar className={classes.toolbar}>
                    <IconButton
                        color="inherit"
                        aria-label="open drawer"
                        onClick={
                            props.state === NavigationPanelStates.open
                                ? props.handleDrawerClose
                                : props.handleDrawerOpen
                        }
                        edge="start"
                        className={classes.menuButton}
                    >
                        {props.state === NavigationPanelStates.open ? (
                            <ChevronLeftIcon />
                        ) : (
                            <MenuIcon />
                        )}
                    </IconButton>
                    <div className={classes.appLogoContainer}>
                        <img
                            className={classes.logo}
                            src="https://s3.amazonaws.com/streamlineathletes.com/assets/programs/22/university-british-columbia_track-field_thunderbirds_logo.png"
                        />
                        <img className={classes.appLogo} src={Logo} />
                    </div>
                    {props.currentUser.permissions.canSeeSearchBar && (
                        <div className={classes.search}>
                            <div className={classes.searchIcon}>
                                <SearchIcon color="secondary" />
                            </div>
                            <Autocomplete
                                id="athlete-select"
                                options={!!props.currentRoster ? props.currentRoster : []}
                                getOptionLabel={(option: Athlete) => option.name}
                                loading={isFetching}
                                classes={{
                                    root: classes.inputRoot,
                                    input: classes.inputInput
                                }}
                                renderOption={(option: Athlete, state: RenderOptionState) => (
                                    <Link className={classes.option} to={profilePath}>
                                        {option.name}
                                    </Link>
                                )}
                                renderInput={params => (
                                    <TextField
                                        {...params}
                                        inputProps={{
                                            ...params.inputProps,
                                            autoComplete: "off"
                                        }}
                                        variant="outlined"
                                        placeholder="Search Athlete..."
                                    />
                                )}
                                autoComplete
                                disableOpenOnFocus
                                onInputChange={onAutocompleteInputChange}
                                open={autocompleteOpen}
                            />
                        </div>
                    )}
                </Toolbar>
            </AppBar>
        </>
    );
}
