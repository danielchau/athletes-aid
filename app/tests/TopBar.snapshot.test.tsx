import * as React from 'react'
import TopBar from '../src/components/TopBar'
import renderer from 'react-test-renderer'

it('InjuryDialog renders correctly', () => {
  const tree = renderer.create(<TopBar state={open} handleDrawerOpen={true} />).toJSON()
  expect(tree).toMatchSnapshot()
})