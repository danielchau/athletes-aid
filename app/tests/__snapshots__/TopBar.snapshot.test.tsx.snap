// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar renders correctly 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed makeStyles-appBar-1 MuiAppBar-colorPrimary mui-fixed"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-4 MuiToolbar-gutters"
  >
    <button
      aria-label="open drawer"
      className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </span>
    </button>
    <div
      className="makeStyles-appLogoContainer-6"
    >
      <img
        className="makeStyles-logo-5"
        src="https://s3.amazonaws.com/streamlineathletes.com/assets/programs/22/university-british-columbia_track-field_thunderbirds_logo.png"
      />
      <img
        className="makeStyles-appLogo-7"
        src={
          Object {
            "process": [Function],
          }
        }
      />
    </div>
    <div
      className="makeStyles-search-8"
    >
      <div
        className="makeStyles-searchIcon-9"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </div>
      <div
        aria-expanded={false}
        aria-owns={null}
        className="MuiAutocomplete-root makeStyles-inputRoot-10"
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="combobox"
      >
        <div
          className="MuiFormControl-root MuiTextField-root"
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
            onClick={[Function]}
          >
            <input
              aria-activedescendant={null}
              aria-autocomplete="both"
              aria-controls={null}
              aria-invalid={false}
              autoCapitalize="none"
              autoComplete="off"
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input makeStyles-inputInput-11 MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
              disabled={false}
              id="athlete-select"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              placeholder="Search Athlete..."
              required={false}
              spellCheck="false"
              type="text"
              value=""
            />
            <button
              aria-label="Clear"
              className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Clear"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="Open popup"
              className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Open popup"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </span>
            </button>
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-135 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-136"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;
